modules/3d/include/opencv2/3d/ptcloud.hpp

```diff
@@ -11,6 +11,334 @@ namespace cv {
 //! @addtogroup _3d
 //! @{
 
+
+//! type of the robust estimation algorithm
+enum SacMethod
+{
+    /** "Random Sample Consensus: A Paradigm for Model Fitting with Applications to Image Analysis and
+     * Automated Cartography", Martin A. Fischler and Robert C. Bolles, Comm. Of the ACM 24: 381–395, June 1981.
+     */
+    SAC_METHOD_RANSAC,
+    //    SAC_METHOD_MAGSAC,
+    //    SAC_METHOD_LMEDS,
+    //    SAC_METHOD_MSAC,
+    //    SAC_METHOD_RRANSAC,
+    //    SAC_METHOD_RMSAC,
+    //    SAC_METHOD_MLESAC,
+    //    SAC_METHOD_PROSAC
+};
+
+enum SacModelType
+{
+    /** The 3D PLANE model coefficients in list [a, b, c, d],
+     corresponding to the coefficients of equation
+     ax + by + cz + d = 0. */
+    SAC_MODEL_PLANE,
+    /** The 3D SPHERE model coefficients in list [center_x, center_y, center_z, radius],
+     corresponding to the coefficients of equation
+     (x - center_x)^2 + (y - center_y)^2 + (z - center_z)^2 = radius^2.*/
+    SAC_MODEL_SPHERE,
+    //    SAC_MODEL_CYLINDER,
+
+};
+
+/**
+ *
+ * @param model_type 3D Model
+ * @return At least a few are needed to determine a model.
+ */
+inline int sacModelMinimumSampleSize(SacModelType model_type)
```

Comment:
Do we really need this function in public API?
Its implementation in public API?

Reply:
For now it seems it is not needed to be public. Have hidden it in `sac_segmentation.cpp`.